custom_components/waste_collection_schedule/waste_collection_schedule/source/krab_se.py

```diff
@@ -0,0 +1,64 @@
+import json
+import urllib
+from datetime import datetime
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Kristianstad Renhållning"
+DESCRIPTION = "Source for Kristianstad Renhållning waste collection."
+URL = "https://renhallningen-kristianstad.se"
+TEST_CASES = {
+    "Case_0": {"street_address": "Östra Boulevarden 1"},
+    "Case_1": {"street_address": "Grenadjärvägen 1"},
+    "Case_2": {"street_address": "Skogsvägen 18"},
+    "Case_3": {"street_address": "Önnestadsvägen 7"},
+}
+
+
+class Source:
+    def __init__(self, street_address):
+        self._street_address = street_address
+
+    def fetch(self):
+        params = [{"query": self._street_address}]
+        headers = {"Module": "universal", 
+                   "Accept": "application/json, text/plain, */*", 
+                   "Unit": "dd905ce7-b16d-4422-be36-564169af4035"}
+        response = requests.get(
+            "https://api-universal.appbolaget.se/waste/addresses/search",
+            headers=headers,
+            params=urllib.parse.urlencode(params),
+        )
```

Comment:
This does not work you cannot `urllib.parse.urlencode` a list, You don't need to encode anything just make your `params` a dict and pass it as `params` the request's library does the converting for you (probably with the same method)